src/features.ts

```diff
@@ -1,44 +1,74 @@
 }
 
-export interface Features {
-  REDmodding: Feature;
-  REDmodLoadOrder: Feature;
-  REDmodAutoconvertArchives: Feature;
+const boolAsFeature = (currentState: boolean): FeatureState =>
+  (currentState ? FeatureState.Enabled : FeatureState.Disabled);
+
+export const IsFeatureEnabled = (featureState: FeatureState): boolean =>
+  featureState === FeatureState.Enabled;
+
+export type Dynamic<T> = () => T;
+
+//
+// Some features can be changed, some can't
+//
+
+export const enum StaticFeature {
+  REDmodding = `v2077-feature-redmodding`,
+  REDmodLoadOrder = `v2077-feature-redmod-load-order`,
+}
+
+export const enum DynamicFeature {
+  REDmodAutoconvertArchives = `v2077-feature-redmod-autoconvert-archives`,
 }
 
-export const CurrentFeatureSet: Features = {
-  REDmodding: Feature.Enabled,
-  REDmodLoadOrder: Feature.Enabled,
-  REDmodAutoconvertArchives: Feature.Enabled,
-};
+export type FeatureSettingsPathInVortex = Record<keyof typeof DynamicFeature, string[]>;
 
-export const DefaultFeatureSetForTesting: Features = {
-  REDmodding: Feature.Disabled,
-  REDmodLoadOrder: Feature.Disabled,
-  REDmodAutoconvertArchives: Feature.Disabled,
-};
 
-export const FeatureEnabled = (feature: Feature): boolean =>
-  feature === Feature.Enabled;
+// FeatureSets are what user code uses...
+
+export type StaticFeatureSet = Record<keyof typeof StaticFeature, FeatureState>;
 
-const boolToEnabled = (currentState: boolean): Feature => (currentState ? Feature.Enabled : Feature.Disabled);
+export type DynamicFeatureSet = Record<keyof typeof DynamicFeature, Dynamic<FeatureState>>;
 
-// @TODO: build features from settings
-export const FeaturesFromSettings = (vortexApi: VortexApi): Features => {
-  const state: types.IState = vortexApi.store.getState();
-  return {
-    REDmodding: CurrentFeatureSet.REDmodding,
-    REDmodLoadOrder: CurrentFeatureSet.REDmodLoadOrder,
-    REDmodAutoconvertArchives: boolToEnabled(util.getSafe(state, [`settings`, `v2077`, `redmod`, `archiveAutoConvert`], false)),
-  };
+type AllFeaturesSet = StaticFeatureSet & DynamicFeatureSet;
+
+export interface FeatureSet extends AllFeaturesSet {
+  fromVersion: string;
+}
+
+
+// ...Through these records
+
+export const FeatureSettingsPath: FeatureSettingsPathInVortex = {
+  REDmodAutoconvertArchives: [...VORTEX_STORE_PATHS.settings, DynamicFeature.REDmodAutoconvertArchives],
 };
+
+export const BaselineFeatureSetForTests: FeatureSet = {
+  fromVersion: `0.9.0`,
+  REDmodding: FeatureState.Disabled,
+  REDmodLoadOrder: FeatureState.Disabled,
+  REDmodAutoconvertArchives: () => FeatureState.Disabled,
+};
+
+export const FeaturesFromSettings = ({ store }: VortexExtensionApi): FeatureSet => ({
+  fromVersion: `0.9.0`,
+  REDmodding: FeatureState.Enabled,
+  REDmodLoadOrder: FeatureState.Enabled,
+  REDmodAutoconvertArchives: () =>
+    boolAsFeature(
+      util.getSafe(store?.getState(), FeatureSettingsPath.REDmodAutoconvertArchives, false),
+    ),
+});
```

Comment:
I like this change a lot. especially the `fromVersion`. It makes sense that if we add more features we specify the version they are introduced in.